pmlc/dialect/pxa/transforms/stencil.cc

```diff
@@ -169,6 +193,66 @@ bool Stencil::ComputeStrideInfo() {
   return tensorsStrides.size() == kNumTensors;
 }
 
+// Collect the non constant indeces used to index the memref at specific index.
+// Ignore indeces that are constant for the ParallelOp.
+BlockArgumentSet Stencil::UsedIdxs(unsigned strideInfoIndex) {
+  assert(strideInfoIndex < kNumTensors);
+
+  BlockArgumentSet used_idxs;
+  for (auto kv : tensorsStrides[strideInfoIndex].strides) {
+    // Make sure the BlockArgument is in the list of the ParallelOp's
+    // BlockArguments.
+    if (opBlockArguments.find(kv.first) != opBlockArguments.end()) {
+      used_idxs.insert(kv.first);
+    }
+  }
+
+  return used_idxs;
+}
+
+void Stencil::PopulateOpBlockArgumentSet() {
+  for (auto blkArg : op.getBody()->getArguments()) {
+    opBlockArguments.insert(blkArg);
+  }
+}
+
+// Collect the indeces that are not constants for the ParallelOp
+// and also the accumulation indeces.
+void Stencil::CollectUsedIndeces() {
+  // The last tensor is the output.
+  tensorIdxs[kNumIndex - 1] = UsedIdxs(kNumIndex - 1);
+  outIdxs = tensorIdxs[kNumIndex - 1];
+  accIdxs.clear();
+  for (unsigned i = 0; i < kNumIndex - 1; ++i) {
+    tensorIdxs[i] = UsedIdxs(i);
+    for (auto idx : tensorIdxs[i]) {
+      if (outIdxs.find(idx) == outIdxs.end()) {
+        accIdxs.insert(idx);
+      }
+    }
+  }
+
+  // Add the out used indeces to the all the used indeces collection as well.
+  allIdxs = accIdxs;
+  allIdxs.insert(outIdxs.begin(), outIdxs.end());
+}
+
+// Get the indeces with stride of one.
+void Stencil::strideOneIdxs(unsigned indx) {
+  for (auto [arg, scale] : tensorsStrides[indx].strides) {
```

Comment:
I think we should probably not rely on C++17 if we plan to upstream this code anytime soon.

Reply:
That is easy to change later and maybe they will themselfs move forward by the time. Either way, changed it. Much less readable, though...